Bottom: 792742960272f8ffa4fe3aef105f7dd780be12e8
Top:    fbc52d2dc6808279c05eeb6c58c5bc566a843627
Author: Ian Kent <raven@themaw.net>
Date:   2013-07-30 18:34:51 +0800

autofs-5.0.7 - try and cleanup after dumpmaps

Try and cleanup a little after dumping maps.
It's not really neccessary but


---

diff --git a/daemon/automount.c b/daemon/automount.c
index 1d0b64e..595d93e 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -2152,6 +2152,17 @@ int main(int argc, char *argv[])
 
 		lookup_nss_read_master(master_list, 0);
 		master_show_mounts(master_list);
+
+		head = &master_list->mounts;
+		p = head->next;
+		while (p != head) {
+			entry = list_entry(p, struct master_mapent, list);
+			p = p->next;
+			master_free_mapent_sources(entry, 1);
+			master_free_mapent(entry);
+		}
+		master_kill(master_list);
+
 		exit(0);
 	}
 
diff --git a/lib/master.c b/lib/master.c
index bbf4b4e..c560496 100644
--- a/lib/master.c
+++ b/lib/master.c
@@ -1427,6 +1427,8 @@ int master_show_mounts(struct master *master)
 			source = source->next;
 		}
 
+		lookup_close_lookup(ap);
+
 		printf("\n");
 	}
